python/adbc_driver_manager: test_lowlevel.py::test_pycapsule failed with PyArrow >= 17

### What feature or improvement would you like to see?

```text
../venv-source/lib/python3.12/site-packages/_pytest/python_api.py:1019: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ExceptionInfo ArrowInvalid('Cannot import released Arrow Stream') tblen=5>
regexp = 'Cannot import released ArrowArrayStream'

    def match(self, regexp: str | Pattern[str]) -> Literal[True]:
        """Check whether the regular expression `regexp` matches the string
        representation of the exception using :func:`python:re.search`.
    
        If it matches `True` is returned, otherwise an `AssertionError` is raised.
        """
        __tracebackhide__ = True
        value = self._stringify_exception(self.value)
        msg = f"Regex pattern did not match.\n Regex: {regexp!r}\n Input: {value!r}"
        if regexp == value:
            msg += "\n Did you mean to `re.escape()` the regex?"
>       assert re.search(regexp, value), msg
E       AssertionError: Regex pattern did not match.
E        Regex: 'Cannot import released ArrowArrayStream'
E        Input: 'Cannot import released Arrow Stream'

../venv-source/lib/python3.12/site-packages/_pytest/_code/code.py:722: AssertionError
=========================== short test summary info ============================
FAILED python/adbc_driver_manager/tests/test_lowlevel.py::test_pycapsule - AssertionError: Regex pattern did not match.
 Regex: 'Cannot import released ArrowArrayStream'
 Input: 'Cannot import released Arrow Stream'
```

https://github.com/apache/arrow-adbc/blob/cda3468e246c5ebbf615b69a83219d075ce78a47/python/adbc_driver_manager/tests/test_lowlevel.py#L457

uses `... ArrowArrayStream` as the expected error message but https://github.com/apache/arrow/pull/40807/files#diff-8897de719613dfdec93c98eca5567d157a0848802cf8554959ca2ebdde3c6839L2329-R2448 (Hmm, we need to "Load diff" of `bridge.cc` explicitly) changed the error message. So the test works with PyArrow < 17 but doesn't work with PyArrow >= 17.

Can we relax the error message pattern or something to support both of old PyArrow and new PyArrow?
Or we may be able to drop support for old PyArrow in our test.

--- a/ci/conda_env_python.txt
+++ b/ci/conda_env_python.txt
@@ -20,13 +20,12 @@ importlib-resources
 # nodejs is required by pyright
 nodejs >=13.0.0
 pandas
-pyarrow=15.0.2
+pyarrow-all
 pyright
 pytest
 setuptools
 
 # For integration testing
-# 0.20.3 is broken on conda-forge
-polars<=0.20.2
+polars
 protobuf
 python-duckdb

--- a/docs/source/python/recipe/postgresql_polars.py
+++ b/docs/source/python/recipe/postgresql_polars.py
@@ -48,9 +48,9 @@ data = pl.DataFrame(
 data.write_database("example", uri, engine="adbc", if_table_exists="replace")
 
 #: After creating the table, we can use
-#: :external:py:func:`polars.read_database` to fetch the result.  Again,
+#: :external:py:func:`polars.read_database_uri` to fetch the result.  Again,
 #: we can just pass the URI and tell Polars to manage ADBC for us.
 
-df = pl.read_database("SELECT * FROM example WHERE ints > 1", uri, engine="adbc")
+df = pl.read_database_uri("SELECT * FROM example WHERE ints > 1", uri, engine="adbc")
 
 assert len(df) == 2

--- a/python/adbc_driver_manager/tests/test_lowlevel.py
+++ b/python/adbc_driver_manager/tests/test_lowlevel.py
@@ -428,7 +428,7 @@ def test_pycapsule(sqlite):
     handle = conn.get_table_schema(catalog=None, db_schema=None, table_name="foo")
     assert data.schema == pyarrow.schema(handle)
     # ensure consumed schema was marked as such
-    with pytest.raises(ValueError, match="Cannot import released ArrowSchema"):
+    with pytest.raises(ValueError):
         pyarrow.schema(handle)
 
     # smoke test for the capsule calling release
@@ -454,7 +454,7 @@ def test_pycapsule(sqlite):
     assert result == table
 
     # ensure consumed schema was marked as such
-    with pytest.raises(ValueError, match="Cannot import released ArrowArrayStream"):
+    with pytest.raises(ValueError):
         pyarrow.table(handle)
 
     # smoke test for the capsule calling release

